marc_to_solr/lib/process_holdings_helpers.rb

```diff
@@ -53,13 +53,24 @@ def permanent_location_code(field_852)
   # Select 876 fields (items) with permanent location. 876 location is equal to the 852 permanent location.
   def select_permanent_location_876(group_876_fields, field_852)
     return group_876_fields if /^scsb/.match?(field_852['b'])
-    group_876_fields.select { |field_876| current_location_code(field_876) == permanent_location_code(field_852) }
+    group_876_fields.select do |field_876|
+      ! in_temporary_location(field_876, field_852)
+    end
   end
 
   # Select 876 fields (items) with current location. 876 location is NOT equal to the 852 permanent location.
   def select_temporary_location_876(group_876_fields, field_852)
     return [] if /^scsb/.match?(field_852['b'])
-    group_876_fields.select { |field_876| current_location_code(field_876) != permanent_location_code(field_852) }
+    group_876_fields.select do |field_876|
+      in_temporary_location(field_876, field_852)
+    end
+  end
+
+  def in_temporary_location(field_876, field_852)
+    # temporary location is any item whose 876 and 852 do not match
```

Comment:
@christinach from what I am hearing from you it is either in a temporary location or it is in a permanent location.  I moved the logic outside the two methods so that we utilize the same for both routines, because it should be in one of the two.  Notice the difference NOT on line 57 vs 65.